server/option.go

```diff
@@ -177,8 +177,12 @@ func (o *scheduleOption) RemoveSchedulerCfg(name string) error {
 			return errors.Trace(err)
 		}
 		if tmp.GetName() == name {
-			schedulerCfg.Disable = true
-			v.Schedulers[i] = schedulerCfg
+			if IsDefaultScheduler(tmp.GetType()) {
+				schedulerCfg.Disable = true
+				v.Schedulers[i] = schedulerCfg
+			} else {
+				v.Schedulers = append(v.Schedulers[:i], v.Schedulers[i+1:]...)
```

Comment:
it seems that it will skip the origin `i+1` item after appending

Reply:
it should be [...)

Reply:
what I mean is in next loop `i` will be `i+1`, but after appending the now `i+1` item is actually the origin `i+2` item, namely it skip origin `i+1` item

Reply:
it will return directly later, no next loop.

Reply:
Oh~ I got it.